backend/src/apiserver/storage/object_store.go

```diff
@@ -40,7 +40,7 @@ type ObjectStoreInterface interface {
 	AddAsYamlFile(o interface{}, filePath string) error
 	GetFromYamlFile(o interface{}, filePath string) error
 	GetPipelineKey(pipelineId string) string
-	GetSignedUrl(bucketConfig *objectstore.Config, secret *v1.Secret, expirySeconds time.Duration, artifactURI string) (string, error)
+	GetSignedUrlWithQueryParams(bucketConfig *objectstore.Config, secret *v1.Secret, expirySeconds time.Duration, artifactURI string, queryParams url.Values) (string, error)
```

Comment:
Same comment as above. Since we are going with a not backwards compatible change here, we might as well keep the name `GetSignedUrl`.